### PR TITLE
Resize charts based on available space

### LINKED ISSUE #4863: Default charts for calibrate
Currently, calibrate plots by default the variables that were selected in the mapping. This is a good start but we should also display:

- [ ]  the error plots for those same variables and 
- [ ] the parameter charts (only those that were not constant)
- [ ] each section (Loss, Parameters, Variables, Errors) should be collapsable

Other enhancements:

- [ ] The output settings panel should be collapsed by default
- [ ] The charts should use the space available

### PR DESCRIPTION
**Pull request by jryu01**
# Description

https://github.com/user-attachments/assets/b9dd0584-2497-40cb-a73f-7af583ef77c3

Charts should resize on simulate and optimize drilldowns as well. 

Resolves #4863

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue [hunk 1]
```diff
@@ -875,7 +876,7 @@ const LOSS_CHART_DATA_SOURCE = 'lossData'; // Name of the streaming data source
 const lossChartRef = ref<InstanceType<typeof VegaChart>>();
 const lossChartSpec = ref();
 const lossValues = ref<{ [key: string]: number }[]>([]);
-const updateLossChartSpec = (data: string | Record<string, any>[]) => {
+const updateLossChartSpec = (data: string | Record<string, any>[], size: { width: number; height: number }) => {
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:879]: It looks to me we are passing `lossChartSize` as a parameter `size` 
Why not just use `lossChartSize.value` for this similar to how you are using `chartSize` sorry if i am missing something obvious

Comment (jryu01) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:879]: Just a preference, I just wanted to make it more explicit with the dependent variables and just wanted it to be decoupled from parent scope although in this case, its simple enough that it doesn't make much difference.

Comment (jryu01) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:879]: It helps in some cases for example, when you look at the watcher

```
watch(
	[() => props.node.state.inProgressCalibrationId, lossChartSize],
	([id, size]) => {
		if (id === '') {
			showSpinner.value = false;
			updateLossChartSpec(lossValues.value, size);
```

By passing the size param (which is watched variable), you can easily tell that `updateLossChartSpec` depends on that variable. If `lossChartSize.value` were used inside the function and size wasn't passed as param here, it wouldn't be clear why `lossChartSize` is being watched here unless you go into and look at the content of the function.

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:879]: This makes total sense!
Is the simulation chart in the same situation as this? If it is are we able to make these two consistent with eachother?

Comment (jryu01) [reply packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue:879]: No simulation drilldown doesn't have this piece of code and the loss chart.

**Review by Tom-Szendrey — APPROVED** [review]

**Review by shawnyama — APPROVED** [review]
neat!

**Pull request merged**